src/components/App.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+import "../styles/App.css";
+import LocationDetails from "./LocationDetails";
+import ForecastDailySummaries from "./ForecastDailySummaries";
+import TopBar from "./TopBar";
+import ForecastMoreDetails from "./ForecastMoreDetails";
+import getForecast from "../requests/getForecast";
+import useUnitContext from "../hooks/useUnitContext";
+import useLocalStorage from "../hooks/useLocalStorage";
+
+function App() {
+  const [location, setLocation] = useLocalStorage("location", "London, GB");
+  const [forecasts, setForecasts] = useState([]);
+  const [selectedDate, setSelectedDate] = useState(0);
+  const [errMessage, setErrMessage] = useState("");
+  const { units } = useUnitContext();
+
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast[0].dateTime === selectedDate
+  );
+
+  const handleCitySearch = (locationInput, unitsInput) => {
+    const selectedLocation = locationInput || location;
+    const selectedUnits = unitsInput || units;
+
+    getForecast(
+      setLocation,
+      setForecasts,
+      setSelectedDate,
+      setErrMessage,
+      selectedLocation,
+      selectedUnits
+    );
+  };
+
+  useEffect(() => {
+    handleCitySearch();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
You shouldnt need to have to use the eslint exception here. If you think about what youre trying to do here, its just to fetch the initial set of forecasts when the page loads for the first time, so you caan just call the `getForecasts` function in here:
```
    getForecast(
      setLocation,
      setForecasts,
      setSelectedDate,
      setErrMessage,
      location,
      units
    );
  }, [location, setLocation, units]);
```